opengrok-indexer/src/main/java/org/opengrok/indexer/search/context/Context.java

```diff
@@ -212,7 +210,7 @@ public boolean getContext2(RuntimeEnvironment env, IndexSearcher searcher,
 
         try {
             List<String> fieldList = qbuilder.getContextFields();
-            String[] fields = fieldList.toArray(new String[fieldList.size()]);
+            String[] fields = fieldList.toArray(new String[0]);
```

Comment:
Why 0?
Above looked like optimization so toArray won't have to allocate new Array, since we know how many elements we will have.
I think if 0 is correct, then in case of above you could just directly call .toArray() with no params

Reply:
IntelliJ recommends 0

Reply:
I remember @ahornace telling me that there is no optimization on using size and 0.

Reply:
Taken directly from idea:
```
There are two styles to convert a collection to an array: either using a pre-sized array (like c.toArray(new String[c.size()])) or using an empty array (like c.toArray(new String[0]).
In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation.
This inspection allows to follow the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs).
```

`.toArray(T[] t)` is necessary so JVM can create array of proper type because of type erasure

Reply:
ok, so this means javadocs suck, I mean, they are outdated :-)